contracts/registry/src/registry.rs

```diff
@@ -406,10 +405,80 @@ impl SBTRegistry for Contract {
     /// Must be called by an SBT contract.
     /// Must emit `Revoke` event.
     /// Must also emit `Burn` event if the SBT tokens are burned (removed).
-    fn sbt_revoke(&mut self, token: Vec<TokenId>, burn: bool) {
+    fn sbt_revoke(&mut self, tokens: Vec<TokenId>, burn: bool) {
         let issuer = env::predecessor_account_id();
-        self.assert_issuer(&issuer);
-        env::panic_str("not implemented");
-        // add events
+        let issuer_id = self.assert_issuer(&issuer);
+        if burn == true {
+            let mut revoked_per_class: HashMap<u64, u64> = HashMap::new();
+            let mut revoked_per_owner: HashMap<AccountId, u64> = HashMap::new();
+            let mut tokens_burned = 0;
+            for token in tokens.clone() {
```

Comment:
don't use clone.

Reply:
```suggestion
            for token in tokens {
```

Reply:
We must `clone()` it since later we are using the value while emitting the event